R/synthetic_cdisc_data.R

```diff
@@ -1,18 +1,43 @@
 #' Get Synthetic CDISC Dataset
 #'
+#' @param archive_name name of data collection. If name = "latest" then the newest dataset gets returned
 #' @param dataset_name the lowercase name of the requested dataset e.g. `adsl`
-#' @param name name of data collection. If name = "latest" then the newest datasets get returned
 #'
 #' @return A data.frame of synthetic data
 #' @export
 #' @examples
 #' \dontrun{
-#' library(scda.2021)
+#' library(scda.2022)
 #'
-#' adsl <- synthetic_cdisc_dataset("adsl", "latest")
+#' adsl <- synthetic_cdisc_dataset("latest", "adsl")
 #' }
-synthetic_cdisc_dataset <- function(dataset_name, name) {
-  synthetic_cdisc_data(name)[[dataset_name]]
+synthetic_cdisc_dataset <- function(archive_name, dataset_name) {
```

Comment:
Changing the order of args is a breaking change:
- have you checked all the packages downstream?
- please add appropriate news entry

Reply:
As an example - `teal.data` would break because of the following: https://github.com/insightsengineering/teal.data/blob/main/R/TealDatasetConnector_constructors.R#L267-L268

Reply:
Good point! I have checked and the only downstream package currently using `synthetic_cdisc_dataset` is `teal.data`, so I have updated the argument name for all uses there and created a PR for this [here](https://github.com/insightsengineering/teal.data/pull/105).

Reply:
great thanks - could you modify NEWS file then to highlight a breaking change?

Reply:
actually there is no differentiation between enhancements and breaking changes so this can stay